test/wordsearch.js

```diff
@@ -0,0 +1,79 @@
+/* global describe, it */
+
+const assert = require('assert');
+const { WordSearchSolver } = require('../');
+
+describe('WordSearchSolver', function () {
+  it('Cardinal directions', function () {
+    const matrix = [
+      [ 'p', 'x', 'x', 'x' ],
+      [ 'u', 'w', 'i', 'n' ],
+      [ 'z', 'x', 'x', 'x' ],
+      [ 'z', 'x', 'x', 'x' ]
+    ];
+    const solver = new WordSearchSolver(matrix);
+    const results = solver.findWords(['puzz', 'win', 'foo', 'bar', 'baz']);
+
+    assert.strictEqual(results.length, 2);
+  });
+
+  it('Diagonal directions', function () {
+    const matrix = [
+      [ 'p', 'x', 'x', 'x', 'b' ],
+      [ 'x', 'u', 'x', 'a', 'x' ],
+      [ 'x', 'x', 'z', 'x', 'x' ],
+      [ 'x', 'x', 'x', 'z', 'x' ]
+    ];
+    const solver = new WordSearchSolver(matrix);
+    const results = solver.findWords(['puzz', 'baz']);
+
+    assert.strictEqual(results.length, 2);
+  });
+
+  it('Jagged array outside box', function () {
+    const matrix = [
+      [ 'x', 'f', 'o', 'o' ],
+      [ 'b', 'x', 'x', ],

```

Comment:
The semistandard linter doesn't seem to like comma dangles and there are three of them in this file.  I pushed a PR to re-enable that linter.

These should be easily fixed with `npx semistandard --fix` from the root of the repo.